File: Python/The3Game.py
class Skipper:
    def __init__(self,skipNum):
        self.skipNum = skipNum

    def next(self,num):
        return self.__findNext__(num)

    def __findNext__(self,num):
        count = num + 1
        while ((str(self.skipNum) not in str(count)) and (count%self.skipNum)) == False:
            count = count + 1
        return count

if __name__ == "__main__":
    print "=== test 3 game === "
    s3 = Skipper(3)
    for i in range(1,50):
        print "num : " +str(i) + " > "+ str(s3.next(i))
    print "=== test 4 game === "
    s4 = Skipper(4)
    for i in range(1,50):
        print "num : " +str(i) + " > "+ str(s4.next(i))

File: Go/The3Game.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func skipper(skip_num int, from int, end int) {
	n := next(skip_num)
	for i := from; i < end; i++ {
		fmt.Printf("num : %d > %d\n", i, n())
	}
}

func next(skip_num int) func() int {
	num := 1
	return func() int {
		num++
		count := num
		for strings.Contains(strconv.Itoa(count), strconv.Itoa(skip_num)) || (count%skip_num) == 0 {
			count++
		}
		return count
	}
}

func main() {
	fmt.Print("=== test 3 game ===\n")
	skipper(3, 1, 10)
	fmt.Print("=== test 4 game ===\n")
	skipper(4, 1, 50)
}
